DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1

```diff
@@ -0,0 +1,177 @@
+Import-Module $PSScriptRoot\..\Helper.psm1 -Verbose:$false
+
+# Localized messages
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData @'
+CheckingZoneMessage          = Checking DNS server zone with name '{0}' is '{1}'...
+AddingZoneMessage            = Adding DNS server zone '{0}' ...
+RemovingZoneMessage          = Removing DNS server zone '{0}' ...
+
+CheckPropertyMessage         = Checking DNS server zone property '{0}' ...
+NotDesiredPropertyMessage    = DNS server zone property '{0}' is not correct. Expected '{1}', actual '{2}'
+SetPropertyMessage           = DNS server zone property '{0}' is set
+'@
+}
+
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]$Name,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]$ZoneFile,
+
+        [ValidateSet('None','NonsecureAndSecure')]
+        [System.String]$DynamicUpdate = 'None',
+        
+        [ValidateSet('Present','Absent')]
+        [System.String]$Ensure = 'Present'
+    )
+
+    $targetResource = @{
+        Name = $Name;
+        ZoneFile = $ZoneFile;
+        DynamicUpdate = $DynamicUpdate;
```

Comment:
This is not finding the current state of Zone's dynamicupdate, but merely coping what user provided. Doesn't sound right?